relay-server/src/actors/outcome_aggregator.rs

```diff
@@ -121,30 +137,38 @@ impl Handler<TrackOutcome> for OutcomeAggregator {
     fn handle(&mut self, msg: TrackOutcome, _ctx: &mut Self::Context) -> Self::Result {
         relay_log::trace!("Outcome aggregation requested: {:?}", msg);
 
-        let TrackOutcome {
-            timestamp,
-            scoping,
-            outcome,
-            event_id,
-            remote_addr,
-            category,
-            quantity,
-        } = msg;
+        if self.mode == AggregationMode::DropEverything {
+            return Ok(());
+        }
+
+        // For lossy aggregation, erase some fields to have fewer buckets
+        let (event_id, remote_addr) = match self.mode {
+            AggregationMode::Lossy => {
+                relay_log::trace!("Erasing event_id, remote_addr for aggregation: {:?}", msg);
+                (None, None)
+            }
+            _ => (msg.event_id, msg.remote_addr),
+        };
+
+        if let Some(event_id) = event_id {
+            relay_log::trace!("Forwarding outcome without aggregation: {}", event_id);
+            self.outcome_producer.do_send(msg).ok();
+            return Ok(());
+        }
 
         // TODO: timestamp validation? (min, max)
-        let offset = timestamp.timestamp() as u64 / self.bucket_interval;
+        let offset = msg.timestamp.timestamp() as u64 / self.bucket_interval;
```

Comment:
I assume that we do not need to validate the timestamp here, because
* outcomes coming in via client reports have their own validation,
* all other outcomes are created by trusted relays.